common/worker_versioning/version_membership_cache.go

```diff
@@ -0,0 +1,28 @@
+//nolint:staticcheck
+package worker_versioning
+
+import enumspb "go.temporal.io/api/enums/v1"
+
+// VersionMembershipCache is used to cache results of Matching's CheckTaskQueueVersionMembership
+// calls (used internally by the worker versioning pinned override validation).
+//
+// Implementations are expected to be safe for concurrent use.
+type VersionMembershipCache interface {
```

Comment:
It feels very clear to me how the cache key, the cache interface, and the cache implementation for events cache are defined right next to each other. Is it possible to do the same for the version membership cache and still achieve the testing that you want? https://github.com/temporalio/temporal/blob/91893f10641bae49f8181235c5e93f61a3c826a8/service/history/events/cache.go#L22-L48